tests/testutils/kubeutils/otlp_sink_deployment.go

```diff
@@ -0,0 +1,285 @@
+// Copyright Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubeutils
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/signalfx/splunk-otel-collector/tests/testutils"
+	"github.com/signalfx/splunk-otel-collector/tests/testutils/kubeutils/manifests"
+)
+
+const sinkName = "otlp-sink"
+
+// OTLPSinkDeployment is a mock Splunk O11Y/Platform backend to capture all
+// cluster telemetry for forwarding to a Testcase's OTLPReceiverSink.
+type OTLPSinkDeployment struct {
+	tc           *testutils.Testcase
+	cluster      *KindCluster
+	sb           *sbRest
+	otlpEndpoint string
+	apiEndpoint  string
+}
+
+func NewOTLPSinkDeployment(cluster *KindCluster) *OTLPSinkDeployment {
+	deployment := &OTLPSinkDeployment{
+		tc:      cluster.Testcase,
+		cluster: cluster,
+	}
+	var err error
+	deployment.sb, err = newSBRest(cluster.Testcase.Logf)
+	require.NoError(deployment.tc, err)
+
+	_, port, err := net.SplitHostPort(deployment.sb.addr)
+	require.NoError(deployment.tc, err)
+	deployment.apiEndpoint = fmt.Sprintf("%s:%s", deployment.cluster.HostFromContainer(), port)
+	deployment.otlpEndpoint = deployment.cluster.OTLPEndointFromContainer()
+
+	deployment.apply()
+	return deployment
+}
+
+func (dep OTLPSinkDeployment) Teardown() {
+	dep.sb.shutdown()
+	so, se, err := dep.cluster.Delete(dep.manifests())
+	require.NoError(dep.tc, err, "stdout: %s, stderr: %s", so.String(), se.String())
+}
+
+func (dep OTLPSinkDeployment) apply() {
```

Comment:
@jinja2 (re: https://github.com/signalfx/splunk-otel-collector/pull/3873#discussion_r1378000551) fwiw I count 163 lines from here to the end of the manifest related bits of this deployment, including the actual `kubectl apply`. The manifests alone in the test file are 165 lines and don't benefit from types, formatting, etc.

Reply:
to be fair, applying the manifests would only add a line or two, so this isn't worth much.